src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can`t be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with such login already exists");
+        }
+        if (user.getAge() < 1) {
+            throw new RuntimeException("Impossible age");
+        }
+        if (user.getAge() < 18) {
+            throw new RuntimeException("Oh my boi, U R 2 young 4 dat");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RuntimeException("Password is too short");
+        }
```

Comment:
Hard to read.
It is better to divide into methods and describe the necessary logic in them.
For example, you can create private methods:
private checkAge(User user) {
// age verification
}